docs/core/deploying/single-file.md

```diff
@@ -0,0 +1,104 @@
+
+Single File deployment is available for both the [framework-dependent deployment model](index.md#publish-framework-dependent) and [self-contained applications](index.md#publish-self-contained). The size of the single file in a self-contained application will be large since it will include the runtime and the framework libraries. The single file deployment option can be combined with [Trim](trim-self-contained.md) and [ReadyToRun](../tools/dotnet-publish.md) publish options.
+
+There are some caveats that you need to be aware for single-file use, chief of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a warning for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the AppHost (the single-file bundle itself) resides. Managed C++ applications are not well suited for single-file deployment and we recommend that you write applications in C# to be single-file compatible.
+
+The single-file option for [self-contained applications](index.md#publish-self-contained) will not bundle native libraries on macOS and Windows machines for .NET 5.0. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. Linux will have all files bundled in .NET 5.0 and will execute the application from the single file. Windows and macOS machines will have an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run.
+
+## Exclude files being embedded in Single File
+
+Certain files can be explicitly excluded from being embedded in the single-file by setting following meta-data:
+
+```xml
+<ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+```
+
+For example, to place some files in the publish directory but not bundle them in the single-file:
+
+```xml
+<ItemGroup>
+    <Content Update="*.xml">
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+      <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+    </Content>
+  </ItemGroup>
+
+```
+
+## Publish a Single File app - CLI
+
+Publish a single file application using the [dotnet publish](../tools/dotnet-publish.md) command. When you publish your app, set the following properties:
+
+- Publish as a self-contained app for a specific runtime: `-r win-x64`
+- Enable trimming: `/p:PublishSingleFile=true`
+
+The following example publishes an app for Windows as a self-contained single file application.
```

Comment:
I find this confusing - the header is "CLI", the above bullet points use command line syntax, but the samples use `csproj`/MSBuild syntax. I know that they do the same thing, but lot of people won't make the connection.

There's also difference in how the SDK behaves in these cases. If the properties are specified in the `csproj` they apply always, so:
* Specifying `RuntimeIdentifier` causes `dotnet build` to output into `/bin/Debug/net5.0/win-x64` (not the default `/bin/Debug/net5.0`.
* Specifying `RuntimeIdentifier` causes `dotnet build` to produce self-contained app (unlike the framework dependent which is the default for normal build)

I personally would not show samples where these settings are in the `csproj`. Because:
* When using CLI - the above weird behaviors will occur
* When using VS the same problems as CLI - and in addition if I use the VS UI to publish it will also create weird interactions - for example the VS UI will not see the values specified in the `.csproj` since it only looks the publish.xml

Reply:
I wanted to encourage users to specify publish settings via a configuration file but I can see that it can be confusing, So changed as you suggested, thanks!